checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -470,6 +473,26 @@ private ExpressionTree getLengthSequenceTree(ExpressionTree lengthTree) {
             return IndexUtil.getLengthSequenceTree(lengthTree, imf, processingEnv);
         }
 
+        /** Infers upper-bound annotation for {@code numerator % divisor}. */
+        private void addAnnotationForRemainder(
+                ExpressionTree numeratorTree,
+                ExpressionTree divisorTree,
+                AnnotatedTypeMirror resultType) {
+            LowerBoundAnnotatedTypeFactory lowerBoundATF = getLowerBoundAnnotatedTypeFactory();
+            UBQualifier result = UpperBoundUnknownQualifier.UNKNOWN;
+            // if numerator >= 0, then numerator%divisor <= numerator
+            if (lowerBoundATF.getAnnotatedType(numeratorTree).hasAnnotation(NonNegative.class)) {
```

Comment:
Positive should be permitted here too. There are methods for this in `LowerBoundTransfer`, but they're private. You could make them accessible or just copy them over, since they're one line each. Or just check for Positive directly here.